cms/djangoapps/maintenance/tests.py

```diff
@@ -0,0 +1,243 @@
+"""
+Tests for the maintenance app views.
+"""
+import ddt
+import itertools
+
+from django.core.urlresolvers import reverse
+from django.utils.translation import ugettext_lazy as _
+from xmodule.modulestore import ModuleStoreEnum
+from xmodule.modulestore.tests.django_utils import ModuleStoreTestCase
+from xmodule.modulestore.tests.factories import CourseFactory, ItemFactory
+from student.roles import GlobalStaff
+from student.tests.factories import AdminFactory, UserFactory
+
+from contentstore.management.commands.utils import get_course_versions
+
+from .views import COURSE_KEY_ERROR_MESSAGES, get_maintenace_urls
+
+
+
+class TestMaintenanceIndex(ModuleStoreTestCase):
+    """
+    Tests for maintenance index view.
+    """
+
+    def setUp(self):
+        """Make the user global staff. """
+        super(TestMaintenanceIndex, self).setUp()
+        self.user = AdminFactory()
+        login_success = self.client.login(username=self.user.username, password='test')
+        self.assertTrue(login_success)
+        self.view_url = reverse('maintenance:maintenance')
+
+    def test_maintenance_index(self):
+        response = self.client.get(self.view_url)
+        self.assertContains(response, "Maintenance", status_code=200)
+
+        # Check that all the expected links appear on the index page.
+        for url in get_maintenace_urls():
+            self.assertContains(response, url, status_code=200)
+
+
+@ddt.ddt
+class MaintenanceViewTestCase(ModuleStoreTestCase):
```

Comment:
use [`SharedModuleStoreTestCase`](http://swampcastle.org/2015/07/28/writing-faster-modulestore-tests.html) since most of these tests don't actually change state. It'll speed it up a lot!